File: src/main/java/cn/labzen/logger/spring/ApplicationRunListener.java
package cn.labzen.logger.spring;

import org.springframework.boot.ConfigurableBootstrapContext;
import org.springframework.boot.SpringApplicationRunListener;
import org.springframework.core.env.ConfigurableEnvironment;

import java.util.Arrays;
import java.util.List;

public class ApplicationRunListener implements SpringApplicationRunListener {

  //private final SpringApplication application;
  //private final String[] args;

  //public ApplicationRunListener(SpringApplication application, String[] args) {
  //  this.application = application;
  //  this.args = args;
  //}

  @Override
  public void environmentPrepared(ConfigurableBootstrapContext bootstrapContext, ConfigurableEnvironment environment) {
    String[] activeProfiles = environment.getActiveProfiles();
    if (activeProfiles.length == 0) {
      Profiles.setCurrentProfiles(List.of(Profiles.DEFAULT));
    } else {
      Profiles.setCurrentProfiles(Arrays.asList(activeProfiles));
    }
  }
}

File: src/main/kotlin/cn/labzen/logger/spring/ApplicationRunListener.kt
package cn.labzen.logger.spring

import org.springframework.boot.ConfigurableBootstrapContext
import org.springframework.boot.SpringApplication
import org.springframework.boot.SpringApplicationRunListener
import org.springframework.core.env.ConfigurableEnvironment

class ApplicationRunListener(
  private val application: SpringApplication,
  private val args: Array<String>
) : SpringApplicationRunListener {

  override fun environmentPrepared(
    bootstrapContext: ConfigurableBootstrapContext,
    environment: ConfigurableEnvironment
  ) {
    if (environment.activeProfiles.isEmpty()) {
      Profiles.currentProfiles = listOf(Profiles.DEFAULT)
    } else {
      Profiles.currentProfiles = environment.activeProfiles.toList()
    }
  }
}
